src/python/pants/bin/goal_runner.py

```diff
@@ -61,6 +65,29 @@ def __init__(
     self._explain = self._global_options.explain
     self._kill_nailguns = self._global_options.kill_nailguns
 
+    if self._parsed_specs.filesystem_specs.dependencies:
+      v1_goals = ' '.join(repr(goal) for goal in self._determine_v1_goals(options))
+      provided_specs = ' '.join(spec.glob for spec in self._parsed_specs.filesystem_specs)
+      approximate_original_command = f"./pants {v1_goals} {provided_specs}"
```

Comment:
Use the `--pants-bin-name` global option instead of hard-coding `./pants`, in case the top-level script is named something else.